### PR TITLE
curriculum: add test in React challenge to detect constructor call

### LINKED ISSUE #39301: Constructor call not checked in React lesson to create component from scratch
**Describe your problem and how to reproduce it:**
In one of the React lessons, _[write a react component from scratch](https://www.freecodecamp.org/learn/front-end-libraries/react/write-a-react-component-from-scratch)_, one of the tasks mentioned is to "Make sure to call the constructor for your component, too." This is not checked in the solution.

The solution that I submitted was:
```
// change code below this line
class MyComponent extends React.Component {
  render() {
    return (
      <div>
        <h1>My First React Component!</h1>
      </div>
    )
  }
}

ReactDOM.render(<MyComponent />, document.getElementById('challenge-node'))
```

**Tell us about your browser and operating system:**
* Browser Name: Google Chrome
* Browser Version:  84.0.4147.89 (64 bit)
* Operating System: Windows 10

### PR DESCRIPTION
**Pull request by Aviel212**
Add a test to check the "MyComponent" constructor's parameters (params) and to check that the super function exists in the constructor.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39301

### CODE REVIEW

**Review by yardenshoham — DISMISSED** [review]

File: curriculum/challenges/english/03-front-end-libraries/react/write-a-react-component-from-scratch.english.md [hunk 1]
```diff
@@ -29,7 +29,8 @@ tests:
     testString: assert((function() { const mockedComponent = Enzyme.mount(React.createElement(MyComponent)); return mockedComponent.find('h1').text() === 'My First React Component!'; })());
   - text: <code>MyComponent</code> should render to the DOM.
     testString: assert(document.getElementById('challenge-node').childNodes.length === 1);
-
+  - text: <code>MyComponent</code> should have a constructor with <code>props</code> calling <code>super</code>.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/03-front-end-libraries/react/write-a-react-component-from-scratch.english.md:32]:
```suggestion
  - text: <code>MyComponent</code> should have a constructor calling <code>super</code> with <code>props</code>.
```

I recommend changing the verbiage here, as "props calling super" doesn't seem clear or accurate. 🙂 The suggested verbiage here comes from [this comment](https://github.com/freeCodeCamp/freeCodeCamp/issues/39301#issuecomment-663088086).

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 . Thank you, for getting this PR out.

@RandellDawson , With this PR, I would be happy with this as is. However, I can see campers getting annoyed at the instructions not telling them to pass `props` from the constructor to `super`. Do you think the instructions need a similar addition?

Currently, the only mention of the constructor in the instructions is:
> Make sure to call the constructor for your component, too.

The fCC-console gives a warning to check the browser console, and the browser console does give very useful errors, if the `super` call is missing.

**Comment by RandellDawson** [comment]
@Sky020 Thinking of how we are currently giving "hints" (targeted tests) in the new project-based curriculum, the hints (similar to the current tests) will give reminders on syntax.  If a previous step in a project introduces a concept or syntax, then following steps may or may not restate the same information, but the hints would be relied upon as the reminder about the syntax.

Applying the same logic here, in a [previous challenge](https://www.freecodecamp.org/learn/front-end-libraries/react/create-a-react-component), the user has already been told about adding the `super` call, so I think it is completely fine to not include it in the description/instructions of this challenge.  Since, the instructions give the helpful hint of "Make sure to call the constructor for your component, too." and the FCC console gives even more hints, I do not think it is necessary to spell things out further.

**Review by naomi-lgbt — APPROVED** [review]
LGTM! Congratulations on submitting your first PR~! 🥳

**Review by RandellDawson — APPROVED** [review]

**Pull request merged**